docs/quick-start.md

```diff
@@ -0,0 +1,251 @@
+# Adding a new element using ProseMirror Elements
+## Description
+The purpose of this document is to give a quick and easy guide to creating an element for a rich text editor using prosemirror-elements.
+
+In this example we will demonstrating this using React with Typescript and utilising the array of functions, types and design patterns that prosemirror-elements gives us for free.
+
+### What is an Element?
+Modelling non-text content in Prosemirror can be tricky. Prosemirror-elements provides a new kind of entity, an Element, that makes this task easier. Elements contain user-defined Fields that can model many different kinds of content, including rich text fields and arbitrary data.
+
+An Element and its Fields are represented as first class citizens of the Prosemirror schema – for example, nested rich text fields play nicely with collaborative editing.
+
+Prosemirror-elements provides a way of expressing an Element and its Fields in the UI framework of your choice, and we provide React bindings as a default.
+What is a Field?
+Each Element is made up of Fields, which represent a data type – for example, text, rich text, or custom data types. 
+
+For example, a Pullquote Element might contain the following fields:
+
+A richText field for styled pullquote content.
+A text field for the attribution.
+A custom field, role, containing type `string` to indicate the pullquote’s presentation.
+A custom field, isMandatory, containing type `boolean` to indicate whether the pullquote should always be displayed.
+# Example: creating a pullquote Element in React
```

Comment:
```suggestion
# Example: creating a pullquote Element in React
```

By convention, we should only have one h1 (`#`) on the page, and everything else should be h2 (`##`) or smaller.